### DISCUSSION TITLE
Questions getting started with brand new installation - Rocky Linux 9 / GO AppImage

### DISCUSSION THREAD

**Discussion by ekortright** [GrandOrgue/grandorgue#1579] [Help] [answered]
I'm trying to upgrade the current GO at our parish church, so I bought a new computer (the current one will become a backup) and installed Rocky Linux 9.2 on it and downloaded the GO 3.12.0-1 AppImage.

I am having a few odd problems with GO:

1. GO does not remember the last organ I opened.  It always opens the demo.  The old version seemed to remember without being told.
2. When GO opens the demo organ the first time, everything works as expected (it just opens the demo organ).  After that, however, every time it opens the demo organ it logs an error to the application log (see below for a sample).  The log just keeps growing and growing unless I remove all GrandOrgue configuration from the user's home folder and start over from scratch, but then GO opens without logging errors only once.  After that the log starts growing again.  I also tried running GO as root, thinking it might be some permissions problem, but the same thing happens: the first time no error is logged, but every time after that the error is logged.  Clearing the errors in the log window clears them only for that session; after exiting GO and running it again, the full list of errors appears again.  As you can see from the log below, the errors only seem to happen when opening the demo organ; if GO could remember the last organ played instead of always opening the demo that would at least hide this problem.
3. I am using a touch screen monitor, and almost everything works perfectly: I can touch organ keys, pistons, pedals, as well as the various +/- buttons in the control window.  However, when I touch a window menu in the control window (File, Audio/Midi, Panel, Help) and the text menu pops up, touching an option makes the option change color for a tiny fraction of a second, but the option is not selected and nothing happens.  When I hover over an option with a mouse, the option changes color as long as the mouse is hovering over it, and clicking with the mouse triggers the menu action.  But touching the menu option does not work, even if I first hover over the option with the mouse and touch it on the screen while it is still highlighted.
 
I am using a Dell P2418HT touch screen monitor with a Lenovo ThinkCentre desktop running Rocky Linux 9.2 (RHEL compatible) with kernel version 5.14.0-284.11.1.el9_2.x86_64.  The desktop is GNOME v. 40.4.0.  The GO AppImage is grandorgue-3.12.0-1.x86_64.AppImage.

Here is a truncated sample of the log file:
> Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: can't open file '/tmp/.mount_grandozw53Nr/usr/share/GrandOrgue/packages
//demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue' (error 2: No such file or directory)
Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: Failed to open '/tmp/.mount_grandozw53Nr/usr/share/GrandOrgue/packages/demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue'
. . .
Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: can't open file '/tmp/.mount_grando2iHKXd/usr/share/GrandOrgue/packages/demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue' (error 2: No such file or directory)
Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: Failed to open '/tmp/.mount_grando2iHKXd/usr/share/GrandOrgue/packages/demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue'
Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: can't open file '/tmp/.mount_grando86Zdau/usr/share/GrandOrgue/packages/demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue' (error 2: No such file or directory)
Fri 16 Jun 2023 09:20:25 PM CDT: 09:20:25 PM: Error: Failed to open '/tmp/.mount_grando86Zdau/usr/share/GrandOrgue/packages/demo-4232D4C616E4B99145537F02B34D342D4E3E3808.orgue'

Can anyone tell me if there is a simple way to get around these problems?  I can try installing GO or even compiling from source instead of using the AppImage, but I thought it might be better to get some advice from the developers before going off in all kinds of directions.  Thanks for any help, and let me know what other information I can provide.

P.S.:  I just realized that Rocky Linux 9 does not use X11, but Wayland.  Is this not yet supported by GO? (GO seems to remember which panels were open and their position, by the way.)  Should I try to switch to X11 and forget about Wayland?
P.P.S.: I switched to X11, but that had no effect on any of the above.

**Comment by oleg68** [comment, marked as answer]
This is an appimage-related problem. It will disappear if you install an .rpm

**Comment by ekortright** [comment]
That's great to hear.  I will install with the .rpm and report back to close out this thread.  Thanks very much for your help.

**Comment by ekortright** [comment]
I downloaded the .rpm file and installed it with `sudo dnf install grandorgue-3.12.1-1.x86_64.rpm`.  In order to install some dependencies not already on this system, I also had to `sudo dnf install epel-release` before installing GO itself.

As you indicated, all of the problems I encountered went away.  Thank you very much!
